R/publish.R

```diff
@@ -0,0 +1,372 @@
+publish_data <- function(dataset) {
+  if (exists("DATAVERSE_SERVER") && exists("DATAVERSE_KEY")) {
+    Sys.setenv("DATAVERSE_SERVER" = DATAVERSE_SERVER)
+    Sys.setenv("DATAVERSE_KEY" = DATAVERSE_KEY)
+    if (require(dataverse)) {
+      library(desc) # this is used to get the author data from the DESCRIPTION file
+      full_dataset <- check_for_existing_id(dataset$name)
+      if (is.list(full_dataset)) {
+        futile.logger::flog.debug("%s: dataverse exists so just update", dataset$name)
+        dataset_id <- full_dataset$datasetId
+        updated_metadata <- generate_new_dataset_metadata(dataset)$datasetVersion
+        update_dataset(dataset = dataset_id, body = updated_metadata)
+        # loop  through the summary dir adding all the files
+        existing_files <- full_dataset$files
+        for (file in dir(dataset$summary_dir)) {
+          file_full_path <- paste0(dataset$summary_dir, "/", file)
+          existing_file_ids <- unique(existing_files[existing_files$filename == file | existing_files$originalFileName == file, "id"])
+          existing_file_id <- existing_file_ids[!is.na(existing_file_ids)]
+          if (length(existing_file_id) > 0) {
+            # allow silent failures - it rejects non-changing updates.
+            futile.logger::flog.debug("submitting file %s", file_full_path)
+            try(update_dataset_file(file = file_full_path, dataset = dataset_id, id = existing_file_id), silent = TRUE)
+          }else {
+            try(futile.logger::ftry(add_dataset_file(file = file_full_path, dataset = dataset_id)), silent = TRUE)
+          }
+        }
+      }else {
+        futile.logger::flog.info("%s: dataverse does not exist so creating a new one", dataset$name)
+        metadata <- generate_new_dataset_metadata(dataset)
+        ds <- create_dataset(dataverse = DATAVERSE_VERSEID, body = metadata)
+        dataset_id <- ds$data$id
+        # loop  through the summary dir adding all the files
+        for (file in dir(dataset$summary_dir)) {
+          try(futile.logger::ftry(add_dataset_file(paste0(dataset$summary_dir, "/", file), dataset_id)), silent = TRUE)
+        }
+      }
+      publish_dataset(dataset_id, minor = FALSE)
+    }else {
+      futile.logger::flog.debug("Dataverse not enabled, no attempt to publish")
+    }
+  }else {
+    futile.logger::flog.debug("No dataverse credentials loaded, no attempt to publish")
+  }
+  return()
+}
+
+check_for_existing_id <- function(dataset_name) {
```

Comment:
This is probably the function you would want to re-use when doing publication - it will convert dataset name into a dataverse dataset and you can then pull the files you need. 
see https://cran.r-project.org/web/packages/dataverse/vignettes/C-retrieval.html for more info on reloading data from R (although I feel like it should be possible without going via disk it might be simplest to just shuffle the csv via the temp directory).

Note that .csv are converted to .tab on dataverse so you would need to either look for original file name, or search for .tab and when you do get_file include format="original" to get it in csv version.